### ISSUE TITLE
Investigate thresholding binary log-loss

### ISSUE THREAD

**Issue by justinormont** [dotnet/machinelearning#5055]
Multi-class log-loss has an `epsilon` value which thresholds the input probability between `epsilon` and `1.0`. This causes multi-class log-loss to never be `Infinity`.

The binary log-loss does not have an `epsilon` threshold, which causes the returned log-loss to be `Infinity` when the prediction is perfectly confident (p=0.0 or p=1.0) and incorrect about the true label.

Log-loss for binary classification:
https://github.com/dotnet/machinelearning/blob/062be280a9d6de23838d4db5ad93fea9d7d0c1f6/src/Microsoft.ML.Data/Evaluators/BinaryClassifierEvaluator.cs#L663-L677

Log-loss for multi-class:
https://github.com/dotnet/machinelearning/blob/062be280a9d6de23838d4db5ad93fea9d7d0c1f6/src/Microsoft.ML.Data/Evaluators/MulticlassClassificationEvaluator.cs#L450-L458

As part of thinking through the right behavior, we should investigate the behavior of other ML packages like scikit-learn/TensorFlow/PyTorch.

**Comment by justinormont** [comment]
The discussion stems from how to handle averaging metrics from cross-validation folds. Making the modification to the evaluator code means that AutoML won't have have to worry about the +/- Infinity cases.

https://github.com/dotnet/machinelearning/pull/5031#discussion_r413252420

**Comment by rosebyte** [comment]
Triage: infinite result might be astonishing, we will take a look but it has low priority.
